### PR TITLE
docs: Add bucket notification documentation.

### LINKED ISSUE #2286: docs: Write comprehensive bucket notification documentation

### PR DESCRIPTION
**Pull request by harshavardhana**
Fixes #2286

### CODE REVIEW

**Comment by harshavardhana** [comment]
@deekoder will be updating this doc soon it is not complete yet, but it is ready for cursory review.

File: docs/bucket/notification/README.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+- _ObjectRemoved_ event is generated whenever an object is deleted.
+
+| Event types| Description|
+|:-----------|:-----------|
+|_s3:ObjectRemoved:*_| All delete operations.|
+|_s3:ObjectRemoved:Delete_||
+
+## Supported Targets
+
+Minio can send bucket notifications to one or more of the following supported targets. The target is identified uniquely through an unique resource identifier called [ARN](http://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html).
```
Comment (krisis) [inline docs/bucket/notification/README.md:28]: We are switching between event notifications and bucket notifications. We should either use one of them everywhere or explain how they could be used interchangeably.

File: docs/bucket/notification/README.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+The Advanced Message Queuing Protocol (AMQP) is an open standard application layer protocol for message-oriented middleware. The defining features of AMQP are message orientation, queuing, routing (including point-to-point and publish-and-subscribe), reliability and security.
+
+Configure `~/.minio/config.json` with your AMQP config.
+
+```json
+	"notify": {
+		"amqp": {
+			"1": {
+				"enable": true,
```
Comment (krisis) [inline docs/bucket/notification/README.md:40]: Can we add a note describe the inputs supplied via config for notification purposes? e.g, how does one choose the values for exchange and routingKey? Are other values permissible?

File: docs/bucket/notification/README.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+				"noWait": false,
+				"autoDeleted": false
+			}
+		},
+        }
+```
+
+### Elasticsearch via Simple Queueing Service (SQS)
+
+Elasticsearch is a search engine based on Lucene. It provides a distributed, multitenant-capable full-text search engine with an HTTP web interface and schema-free JSON documents. To know more details about configuration please read [Minio server configuration guide](https://docs.minio.io/docs/minio-server-configuration-files-guide)
```
Comment (krisis) [inline docs/bucket/notification/README.md:58]: The link to server configuration guide should appear in other targets' documentation too. It is missing in AMQP section.

File: docs/bucket/notification/README.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+			}
+		},
+        }
+```
+
+### Listen via Simple Notification Service (SNS)
+
+Applications may directly listen for events using an extended [ListenBucketNotification RESTful API](https://docs.minio.io/docs/golang-client-api-reference#ListenBucketNotification).
+
+You may also look at [mc watch](https://docs.minio.io/docs/minio-client-complete-guide#watch) command to write useful shell scripts around bucket events.
```
Comment (krisis) [inline docs/bucket/notification/README.md:97]: We are calling event-notifications by different names within this document. We need a uniform convention for this.

**Review by krisis — CHANGES_REQUESTED** [review]

**Comment by harshavardhana** [comment]
This document is not ready for review @krisis .

**Pull request closed without merging**